Add brew-cask-zsh-completion

joshka opened the pull request:
This was previously removed from homebrew-completions, but based on a few
discussions at https://github.com/Homebrew/brew/pull/407 this belongs in this
repo for now. The completions update those currently found in the Oh My Zsh
plugin. See https://github.com/joshka/brew-cask-zsh-completions for more info.

joshka commented:
The fails on this are:

> homebrew/completions/brew-cask-zsh-completion:
>   * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
>   * GitHub repository too new (<30 days old)

I can understand how this policy might be effective to ensure quality / security etc. One of the homebrew devs suggested pulling the source repo into github/homebrew. Would that do the trick?

bfontaine commented:
> One of the homebrew devs suggested pulling the source repo into github/homebrew.

Do you mind posting a link to the relevant comment?

> Would that do the trick?

These checks are for `homebrew/core`; we’re more tolerant here.

joshka commented:
>>One of the homebrew devs suggested pulling the source repo into github/homebrew.

>Do you mind posting a link to the relevant comment?

https://github.com/Homebrew/brew/pull/407#issuecomment-230081558

joshka commented:
Homebrew/brew#407 was merged last night, and I've requested that the removal of the plugin from oh-my-zsh. What would you like to do with this PR? Can we merge it as is?

bfontaine commented on brew-cask-zsh-completion.rb:
```diff
@@ -0,0 +1,18 @@
+
+  test do
+    assert_match "_brew_cask",
+      shell_output("zsh -cf 'fpath+=/usr/local/share/zsh/site-functions; autoload -U compinit && compinit; whence _brew_cask'")
```
Please use `#{zsh_completion}` instead of `/usr/local/share/zsh/site-functions` so that it’ll work with any Homebrew prefix.

bfontaine commented:
@joshka Just one little nit for the test and I think we can merge.

joshka commented:
@bfontaine I've updated the PR (and squashed the commits).
That said, perhaps https://github.com/Homebrew/brew/pull/936 is the better approach for this now that brew-cask is part of Homebrew core. This is consistent with some of the discussion at https://github.com/Homebrew/brew/pull/407, and I'd be happy for you to merge either this PR or https://github.com/Homebrew/brew/pull/936

joshka commented:
https://github.com/Homebrew/brew/pull/936 is merged making this now redundant.
